test/test_testing.py

```diff
@@ -4073,6 +4073,228 @@ def test_add_layer_merge(self, client: _TestingPebbleClient):
             == plan.to_yaml()
         )
 
+    def test_add_layer_checks_combine_override_replace(self, client: _TestingPebbleClient):
+        plan = client.get_plan()
+        assert isinstance(plan, pebble.Plan)
+        assert plan.to_yaml() == '{}\n'
+        client.add_layer(
+            'foo',
+            pebble.Layer("""\
+            checks:
+                up:
+                    override: replace
+                    level: alive
+                    period: 30s
+                    threshold: 1
+                    exec:
+                        command: service nginx status
+            """),
+        )
+        plan = client.get_plan()
+        # The YAML should be normalized
+        assert (
+            textwrap.dedent("""\
+            checks:
+              up:
+                exec:
+                  command: service nginx status
+                level: alive
+                override: replace
+                period: 30s
+                threshold: 1
+            """)
+            == plan.to_yaml()
+        )
+        client.add_layer(
+            'foo',
+            """\
+            checks:
+                up:
+                    override: replace
+                    level: alive
+                    period: 10s
+                    threshold: 5
+                    exec:
+                        command: service nginx status
+            """,
+            combine=True,
+        )
+        plan = client.get_plan()
+        assert (
+            textwrap.dedent("""\
+            checks:
+              up:
+                exec:
+                  command: service nginx status
+                level: alive
+                override: replace
+                period: 10s
+                threshold: 5
+            """)
+            == plan.to_yaml()
+        )
+
+    def test_add_layer_checks_combine_override_merge(self, client: _TestingPebbleClient):
+        client.add_layer(
+            'foo',
+            """\
+            checks:
+                up:
+                    level: alive
+                    period: 30s
+                    threshold: 1
+                    exec:
+                        command: service nginx status
+                ready:
+                    level: ready
+                    period: 30s
+                    threshold: 1
+                    exec:
+                        command: service nginx ready
+            """,
+        )
+        client.add_layer(
+            'foo',
+            """\
+            checks:
+              up:
+                override: merge
+                level: alive
+                period: 10s
+                threshold: 5
```

Comment:
Here's where we need significantly more test cases. Perhaps have one `exec` check, one `http`, and one `tcp`. And may need a test where the base layer doesn't have `exec.environment` and one where it does, so we're testing all the different cases in our new logic. Similar with `http.headers`.